api/iot/v1beta1/iot_helpers.go

```diff
@@ -0,0 +1,56 @@
+package iot
+
+import (
+	"time"
+
+	"github.com/scaleway/scaleway-sdk-go/internal/async"
+	"github.com/scaleway/scaleway-sdk-go/internal/errors"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+const (
+	waitForHubDefaultTimeout = 15 * time.Minute
+	defaultRetryInterval     = 5 * time.Second
+)
+
+// WaitForHubRequest is used by WaitForHub method.
+type WaitForHubRequest struct {
+	HubID         string
+	Region        scw.Region
+	Status        HubStatus
+	Timeout       *time.Duration
+	RetryInterval *time.Duration
+}
+
+// WaitForHub waits for the hub to be in a ready state before returning.
+func (s *API) WaitForHub(req *WaitForHubRequest) (*Hub, error) {
+	timeout := *req.Timeout
```

Comment:
`*req.Timeout` might be nil

```suggestion
	timeout := defaultTimeout
	if req.Timeout != nil {
		timeout = *req.Timeout
	}
```

Reply:
You're completely right, this is even done on the next parameter check. Fixed, thanks!